Add Address Functionality

jh-206 opened the pull request:
Extracts .mapaddress from Craigslist rental ad. This requires reading html page for each posting, which increases the run-time and number of web pages to read. Therefore, included is logical argument to get_query function ( get_address, default is FALSE) on whether or not to extract address. If TRUE, Address column added to output dataframe. If FALSE, output same as before.

Ellipses added to rentals function so the arguments can be passed to get_query.
Use example:
```
rentals(location = "seattle", get_address = TRUE)
```

Commit:
Add Address Functionality

Extracts `.mapaddress` from rental ad. This requires reading html page for each posting, so included is logical argument to `get_query` function on whether or not to extract address.

Commit:
Update: Documentation for rentals function argument ellipses

Commit:
Update: Updated documentation

Ran devtools::document() to update

plnnr commented:
This is great! Do you know how I can make `get_address = True` when I'm calling the `rentals`  function?

Edit: Maybe I have to wait for the latest code to commit.

jh-206 commented:
`rentals` function now has ellipses that passes arguments to `get_query`
So you can just include in the rentals function arguments `rentals(get_address = TRUE)`

plnnr commented:
I'm kinda dumb with github and R. I just did `devtools::installl_github("adletaw/craigr")` and it says "skipping install... the SHA1 (ca2f30c9) has not changed since last install..."
But when I run this line `results <- craigr::rentals(location = "seattle", area = "see", get_addresses = TRUE, max_results = 100)` it says that `get_addresses = TRUE` is an unused argument.

jh-206 commented:
The pull request has not been merged by @adletaw, so the repository you installed has none of these changes

plnnr commented:
Ok, I figured it was something like that. So I just wait til they merge/commit the changes?

jh-206 commented:
It's not clear whether @adletaw will want to merge this pull request or if they are even actively maintaining this repository

If you want to test or use, you could install the code from my fork of the repository
`devtools::install_github("jh-206/craigr")`

Commit:
Bug Fix: Handle zero search results

Function `get_num_results` returns a `numeric(0)` when there are zero search results. This value is then compared logically to `max_results`, breaking the code. Modified so `get_num_results` returns the number `0` instead

Commit:
Update: Return NA for no search results

Commit:
Revert "Update: Return NA for no search results"

This reverts commit 07d97aa1db4a56c1bd7ca61f2b002d7f65db463f.

Commit:
Revert "Bug Fix: Handle zero search results"

This reverts commit 4fdfaf6cf673b04dc1d8260d827ba0b834718fd7.

jh-206 commented:
As this repository does not appear to be actively maintained, I am creating a branch in my fork of the repository `jh-206/craigr` called `active`, and setting that branch as my default.

The `master` branch of my fork of the repository will remain identical to the code used to generate this pull request, while the `active` branch will be changed moving forward.

jh-206 commented:
For active branch, use `devtools::install_github("jh-206/craigr", ref = "active")`

plnnr commented:
I still haven't been able to get it to work unfortunately. I did `devtools::install_github("jh-206/craigr", ref = "active")` but when I call `results <- craigr::rentals(location = "seattle", area = "see", get_addresses = TRUE, max_results = 100)` it comes back with `unused argument (get_addresses = TRUE)`. I also tried `remove.packages("craigr")` in case the old installation was giving me issues.

jh-206 commented:
`results <- craigr::rentals(location = "seattle", area = "see", get_address = TRUE, max_results = 100)`

`get_address` rather than `get_addresses`

plnnr commented:
WOW, thank you :)
